python/lsst/ts/observatory/control/remote_group.py

```diff
@@ -1207,3 +1207,47 @@ async def __aenter__(self: TRemoteGroup) -> TRemoteGroup:
 
     async def __aexit__(self, *args: typing.Any) -> None:
         await self.close()
+
+    async def show_auth_status(self) -> None:
+        """Shows the authorization status of a group"""
+
+        async def _log_auth_status(c: salobj.Remote) -> None:
+            """Helper function to log authorization status"""
+            authList = await c.evt_authList.aget()
+            self.log.info(
+                f"private_identity: {authList.private_identity}, "
+                f"authorizedUsers: {authList.authorizedUsers}, "
+                f"nonAuthorizedCSCs: {authList.nonAuthorizedCSCs}"
+            )
+
+        await asyncio.gather(
+            *[
+                _log_auth_status(getattr(self.rem, c))
+                for c in self.components_attr
+                if getattr(self.rem, c) is not None
+            ]
+        )
+
+    async def assert_user_is_authorized(
+        self, user: typing.Optional[str] = None
+    ) -> None:
+        """Asserts that the (current) user is authorized to command all the CSCs in the current group.
+
+        Parameters
+        ----------
+        user : str or None, Default: None
+            The user that should have the authorization to command all the CSCs in the group.
+        """
+        identity = user if user else self.get_identity()
+        
+        async def _assert_user_is_authorized(c: salobj.Remote) -> None:
```

Comment:
Same comment as above, I think you could use `_aget_topic_samples_for_components` and avoid having these internal methods.